edb/ir/ast.py

```diff
@@ -511,6 +511,9 @@ class TypeRoot(Expr):
     # This will be replicated in the enclosing set.
     typeref: TypeRef
 
+    # Whether this is a reference to a cached global.
+    is_cached_global: bool = False
```

Comment:
Cached global?

Do we actually cache the value, or is this just your way of saying "a global with an assigned expression whose value we compute at runtime"?

Reply:
Oh, cached in the query, right.

I though it was cached in edgedb connection or even edgedb database.